kmip/tests/unit/core/messages/payloads/test_sign.py

```diff
@@ -0,0 +1,752 @@
+
+        self.assertEqual(None, payload.unique_identifier)
+        self.assertEqual(None, payload.signature_data)
+
+    def test_init_with_args(self):
+        """
+        Test that a Sign response payload can be constructed with valid
+        values.
+        """
+        payload = sign.SignResponsePayload(
+            unique_identifier='00000000-1111-2222-3333-444444444444',
+            signature_data=b'\x01\x02\x03'
+        )
+
+        self.assertEqual(
+            '00000000-1111-2222-3333-444444444444',
+            payload.unique_identifier
+        )
+        self.assertEqual(b'\x01\x02\x03', payload.signature_data)
+
+    def test_invalid_unique_identifier(self):
+        """
+        Test that a TypeError is raised when an invalid value is used to set
+        the unique identifier of a Sign response payload.
+        """
+        payload = sign.SignResponsePayload()
+        args = (payload, 'unique_identifier', 0)
+        self.assertRaisesRegexp(
+            TypeError,
+            "unique identifier must be a string",
+            setattr,
+            *args
+        )
+
+    def test_invalid_signature_data(self):
+        """
+        Test that a TypeError is raised when an invalid value is used to set
+        the signature data of a Sign response payload.
+        """
+        payload = sign.SignResponsePayload()
+        args = (payload, 'signature_data', 0)
+        self.assertRaisesRegexp(
+            TypeError,
+            "signature data must be bytes",
+            setattr,
+            *args
+        )
+
+    def test_read(self):
+        """
+        Test that a Sign response payload can be read from a data stream.
+        """
+        payload = sign.SignResponsePayload()
+
+        self.assertEqual(None, payload.unique_identifier)
+        self.assertEqual(None, payload.signature_data)
+
+        payload.read(self.full_encoding)
+
+        self.assertEqual(
+            'b4faee10-aa2a-4446-8ad4-0881f3422959',
+            payload.unique_identifier
+        )
+        self.assertEqual(
+            b'\x01\x02\x03\x04\x05\x06\x07\x08'
+            b'\x09\x10\x11\x12\x13\x14\x15\x16',
+            payload.signature_data
+        )
+
+    def test_read_invalid(self):
+        """
+        Test that a ValueError gets raised when required Sign response
+        payload attributes are missing from the payload encoding.
+        """
+        payload = sign.SignResponsePayload()
+        args = (self.empty_encoding, )
+        self.assertRaisesRegexp(
+            ValueError,
+            "invalid payload missing the unique identifier attribute",
+            payload.read,
+            *args
+        )
+
+        payload = sign.SignResponsePayload()
+        args = (self.incomplete_encoding, )
+        self.assertRaisesRegexp(
+            ValueError,
+            "invalid payload missing the signature data attribute",
+            payload.read,
+            *args
+        )
+
```

Comment:
The `read` tests look good but there need to be corresponding `write` tests as well.